tasks/hyrax-metadata-updates/index.js

```diff
@@ -236,6 +236,37 @@ async function getMetadataObject(metadataFileName, metadata) {
   return { metadataObject, isUmmG };
 }
 
+/**
+ * isAppropriateForHyrax
+ * Find the suitable formats in config.
+ * If the config doesn't exist, assume that any granule in this collection
+ * is for Hyrax and return true.
+ * If the config exists and the granule doesn't define a format then
+ * return false.
+ * If the config exists and the format of the granule does not match on of the
+ * configured formats then return false.
+ *
+ * @param {Object} config - comnfiguration
+ * @param {Object} metadata - the granule metadata
+ * @param {boolean} isUmmG - whether this is UMM-G or ECHO10 metadata
+ * @returns {boolean} whether the granule metadata needs a Hyrax URL added to it or not
+ */
+function isAppropriateForHyrax(config, metadata, isUmmG) {
+  let appropriate = true;
+  const formats = get(config.hyrax, 'formats');
```

Comment:
`hyrax` isn't a required property of `config`, so I think this could also throw an error. Updating to `get(config, 'hyrax.formats');` should work

I might suggest this broader refactor which is a little easier to read, but take/leave as you see it

```
function isAppropriateForHyrax(config, metadata, isUmmG) {
  const formats = get(config, 'hyrax.formats');
  if (isUndefined(formats)) {
    return true;
  }

  const format = isUmmG 
    ? get(metadata, 'DataGranule.ArchiveAndDistributionInformation[0].Format')
    : get(metadata, 'Granule.DataFormat')
  
  return formats.includes(format);
}
```

Reply:
The refactor looks fine to me except we are no longer catching the exception thrown if the non-mandatory elements we are looking for are not there, thus marking the granule inappropriate for metadata update.

Reply:
If `DataGranule.ArchiveAndDistributionInformation[0].Format` does not exist, then `get(metadata, 'DataGranule.ArchiveAndDistributionInformation[0].Format')` should return `undefined`. And thus I think `formats.includes(undefined)` would evaluate to `false`, which has the same effect as the try/catch. But I assume the unit tests would verify whether that is correct

Reply:
Why yes! They do verify that. Cheers.

Reply:
Commited